File: Python/Easy/125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if s == '':
            return True
        s = ''.join(ch for ch in s if ch.isalnum()).lower()
        left, right = 0, len(s) - 1
        while left < right:
            if s[left] == s[right]:
                left += 1
                right -= 1
            else:
                return False
        return True



# sol = Solution()
# sol.isPalindrome(s = "  Hello,  World! 123  ")
# print(sol)

File: C++/Easy/125.cc
#include <string>
#include <cctype>
#include <iostream>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        string t;
        t.reserve(s.size());
        for (char c : s){
            if (isalnum(c)){
                t.push_back(tolower(c));
            }
        }
        int l{0}, r{(int)t.size() - 1};
        while (l < r){
            if (t[l] == t[r]){
                l++;
                r--;
                continue;
            }
            else {
                return false;
            }
        }
        return true;

    }
};